native/cocos/bindings/jswrapper/State.cpp

```diff
@@ -27,24 +27,4 @@
 #include "State.h"
 
 namespace se {
-
-State::State(Object *thisObject)
-: _thisObject(thisObject) {
-    if (_thisObject != nullptr) {
-        _thisObject->incRef();
-    }
-}
-
-State::State(Object *thisObject, const ValueArray &args)
-: _thisObject(thisObject),
-  _args(&args) {
-    if (_thisObject != nullptr) {
-        _thisObject->incRef();
-    }
-}
-
-State::~State() {
-    SAFE_DEC_REF(_thisObject);
-}
```

Comment:
Any reason to implement it in .h?

Reply:
The code size of this function is somewhats small and `se::State` objects are only constructed and destructed within `jsb*Wrapper` functions, which are not frequently used.  By inlining, the destructor fewer function calls will be generated within JSB functions.

Reply:
Yep, the code size is small. But if there is not special reason, then should implement in .cp by default.

Reply:
Yes, for most cases destructors are implemented in source files. Here these `jsb*Wrapper` functions are hot and inlined.

Reply:
Ok

Reply:
Please do testing about it. We should do decision base on testing result. If it is true, then it is a new rule, we should check other source code too.

Reply:
### Code size
||final code size|
|--|----|
|inline| 21365912 B (+208B)|
|non-inline|21365704 B|

### instructions

### non-inline
![image](https://user-images.githubusercontent.com/40414978/191908992-f131fdaa-2e8c-4627-a1bf-a7c68fc5757d.png)
![image](https://user-images.githubusercontent.com/40414978/191911861-8616bffd-3dda-468d-a0be-277cb796cc31.png)

#### inline
![image](https://user-images.githubusercontent.com/40414978/191908945-547385bc-100c-4297-ae81-9dc6cbc5a2aa.png)

less instructions are emitted, and should be a bit faster.